packages/components/atoms/f-filter-pill/src/components/FilterPill.vue

```diff
@@ -1,29 +1,183 @@
     </div>
 </template>
 
 <script>
+import { TickIcon } from '@justeat/f-vue-icons';
+
 export default {
     name: 'FilterPill',
-    components: {},
+    components: {
+        TickIcon
+    },
     props: {
+        inputId: {
+            type: String,
+            default: null
+        },
+        toggleValue: {
+            type: String,
+            default: null
+        },
+        isSelected: {
+            type: Boolean,
+            default: false
+        },
+        isDisabled: {
+            type: Boolean,
+            default: false
+        },
+        displayText: {
+            type: String,
+            default: null
+        },
+        displayNumber: {
+            type: Number,
+            default: null
+        }
+    },
+    data () {
+        return {
+            isToggleSelected: false,
+            isToggleDisabled: false
+        };
+    },
+    computed: {},
+    watch: {
+        isSelected (newValue) {
+            if (newValue !== this.isToggleSelected) {
+                this.isToggleSelected = newValue;
+            }
+        },
+        isDisabled (newValue) {
+            if (newValue !== this.isToggleDisabled) {
+                this.isToggleDisabled = newValue;
+            }
+        }
+    },
+    mounted () {
+        this.isToggleSelected = this.isSelected;
+        this.isToggleDisabled = this.isDisabled;
+    },
+    methods: {
+        toggleFilter () {
+            this.isToggleSelected = !this.isToggleSelected;
+
+            this.$nextTick(() => {
+                this.$emit('toggle', this.toggleValue);
+            });
+        }
     }
 };
 </script>
 
 <style lang="scss" module>
+$filter-pill-selected: $color-content-positive;
+$filter-pill-checkbox-width: 16px;
+
+@mixin ellipsis() {
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+}
+
 .c-filterPill {
+    outline: 1px solid $color-grey-30;
+    border-radius: $radius-rounded-e;
+    background: $color-white;
+
+    &:focus-within:not(.c-filterPill--disabled) {
+        outline: 2px solid $color-focus;
+    }
+}
+
+.c-filterPill--selected {
+    outline-color: $filter-pill-selected;
+}
+
+.c-filterPill-label {
     display: flex;
-    justify-content: center;
-    min-height: 80vh;
-    width: 80vw;
-    margin: auto;
-    border: 1px solid red;
+    padding: spacing(b) spacing(d);
+    cursor: pointer;
+    @include font-size(body-m);
     font-family: $font-family-base;
-    @include font-size(heading-m);
+    font-weight: $font-weight-regular;
+
+    &:focus {
+        outline: none; // Prevents Safari doubling focus styles.
+    }
+
+    .c-filterPill--selected & {
+        color: $filter-pill-selected;
+    }
+
+    .c-filterPill--disabled & {
+        color: #efedea;
+        cursor: default;
+    }
+}
+
+.c-filterPill-icon {
+    align-self: center;
+    opacity: 0;
+    width: 0;
+
+    .c-filterPill--selected &, .c-filterPill:hover:not(.c-filterPill--disabled) & {
+        opacity: 1;
```

Comment:
This is super minor & maybe personal preference but I would but the two selectors on different lines for added readability.